chicken-sloths/bangazon-cli-sprint2
# Customers Controller (and tests!)

# Description
This PR hooks up the Controller for the add customer route. Now you should be able to select option `1` from the main menu, be prompted for customer input, and post a new customer to the database. 

## Related Ticket(s)
I *think* this should fix Issue #1

## Problem to Solve
We had all the right data going to and from the database in customers model, and we had all the prompts set up, but we hadn't connected the two yet!

## Proposed Changes
Builds out `promtNewCustomer()` function in `createCustC.js` to check for duplicate customers in the DB before we post and, if the customer doesn't already exist, posts it to the db.

## Expected Behavior and Steps to Test
1. `npm start`
1. Select option 1
1. Enter the following customer data as prompted:
```
      first_name: Nicolas,
      last_name: Cage,
      account_creation_date: The dawn of time,
      street_address: Rock and Roll Road,
      city: Nashville,
      state: TN,
      postal_code: 37217,
      phone_number: 888-888-8888
```
1. You should get a success message with the id of the customer you just entered (it should be 50)
1. Open your database in your browser of choice and you should see St. Nic down there at the bottom.
1. Try entering a customer that's already in the database! Here's Rosie. 

(If you want to be an extra thorough tester, feel free to enter all of Rosie's info. If you're feeling kind of lazy, you can just enter any random string for everything EXCEPT her phone number. Just copy and paste her phone number in there.)

```
      first_name: 'Rosie',
      last_name: 'Waters',
      account_creation_date: '2018-04-05T07:56:00.279Z',
      street_address: '4763 Kenny Turnpike',
      city: 'Huberttown',
      state: 'Georgia',
      postal_code: '17176',
      phone_number: '792.555.3469 x367'
```
1. You should get an error message saying that the customer already exists.
1. Run ``npm test`` and all tests should pass.


## Testing

- [ x] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
- [ x] I certify that all existing tests pass

## Documentation

## A note about testing for duplicates
I really wanted to find a way to test the *feature* of preventing duplicates when adding a new customer, but that feature is wrapped up in the `promtNewCustomer()` function, which I can't figure out how to test as a whole because it involves getting prompt input. Instead, I tested the feature of getting a single customer by their phone number because if that works, it should also prevent you from adding that same customer. Does that make sense? If this is confusing I can explain more, but I'm trying to live Joe's advice of testing features not functions.
